[2.0] Release notes and highlights for 2.0.0 (#5427)
elastic/cloud-on-k8s: docs/release-notes/highlights-2.1.0.asciidoc

[[release-highlights-2.1.0]]
== 2.1.0 release highlights

[float]
[id="{p}-210-new-and-notable"]
=== New and notable

> New and notable changes in version 2.1.0 of {n}. See <<release-notes-2.1.0>> for the full list of changes.

Reviewer: ```suggestion
New and notable changes in version 2.1.0 of {n}. Check <<release-notes-2.1.0>> for the full list of changes.
```
Avoid "see" for accessibility reasons.

[float]
[id="{p}-210-disabling-upgrade-predicates"]
==== Allowing upgrade predicates to be selectively disabled

> Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearcch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process.  Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details. 

Author: I moved this to the bottom of the highlights, and noted the danger.  I'll wait on @alaudazzi to see if it's preferred to simply remove it.

> Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearcch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process.  Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details. 

Reviewer: Sorry for the late reply. The emphasis given on the fact that it's _a troubleshooting tool of last resort_ is very clear.

> Additional information was added to the Elasticsearch status sub-resource, which provides rich details into the in-progress operations during upgrades, upscale, and downscale operations.  New conditions fields include `ReconciliationComplete`, `RunningDesiredVersion`, and `ElasticsearchIsReachable` which give details into why each condition is either True, or False.  Also included is a new parent field `inProgressOperations`, which provides detailed topology information for upgrades, upscale, and downscale operations.

Reviewer: ```suggestion
Additional information was added to the Elasticsearch status sub-resource, which provides rich details into the in-progress operations during upgrades, upscale, and downscale operations. New conditions fields include `ReconciliationComplete`, `RunningDesiredVersion`, and `ElasticsearchIsReachable` which give details into why each condition is either True, or False. Also included is a new parent field `inProgressOperations`, which provides detailed topology information for upgrades, upscale, and downscale operations.
```

> Additional information was added to the Elasticsearch status sub-resource, which provides rich details into the in-progress operations during upgrades, upscale, and downscale operations. New conditions fields include `ReconciliationComplete`, `RunningDesiredVersion`, and `ElasticsearchIsReachable` which give details into why each condition is either True, or False. Also included is a new parent field `inProgressOperations`, which provides detailed topology information for upgrades, upscale, and downscale operations.

Reviewer 2: "provide details into" sounds off to me but I am no native speaker. 

Also we have 2 x "details into" and then "detailed topology information" in one paragraph which sound to me like too much detail 😄

Author: I've made some adjustments to the language here.  Let me know if it's more readable.

[float]
[id="{p}-210-ES-Kibana-set-status-observed-generation"]
==== Improved Elasticsearch and Kibana generation status

> An additional field `observedGeneration` is now maintained within Elasticsearch and Kibana's status sub-resource.  This new field represents the current generation of the specification that the ECK operator is working to reconcile, and is intended to allow tools to deterministically monitor the rollout of custom resources.

Reviewer: ```suggestion
An additional field `observedGeneration` is now maintained within Elasticsearch and Kibana's status sub-resource. This new field represents the current generation of the specification that the ECK operator is working to reconcile, and is intended to allow tools to deterministically monitor the rollout of custom resources.
```

> A welcome improvement in ECK 2.1 are additions to the Elasticsearch status sub-resource, which provide rich details into the in-progress operations during upgrades, upscale, and downscale operations.  New conditions fields include `ReconciliationComplete`, `RunningDesiredVersion`, and `ElasticsearchIsReachable` which give details into why each condition is either True, or False.  Also included is a new parent field `inProgressOperations`, which provides detailed topology information for upgrades, upscale, and downscale operations.

Author: Reworded.   Makes complete sense.

> A welcome improvement in ECK 2.1 are additions to the Elasticsearch status sub-resource, which provide rich details into the in-progress operations during upgrades, upscale, and downscale operations.  New conditions fields include `ReconciliationComplete`, `RunningDesiredVersion`, and `ElasticsearchIsReachable` which give details into why each condition is either True, or False.  Also included is a new parent field `inProgressOperations`, which provides detailed topology information for upgrades, upscale, and downscale operations.

Reviewer: Sorry for the late reply. The reworded text LGTM.

[float]
[id="{p}-210-bugfixes"]
==== Bug fixes

> Several bugs have been fixed in this release, including Elasticsearch status updates during errors, and consistently handling upgrades of non-HA (2 nodes) Elasticsearch clusters.

Reviewer 2: I would remove this from the highlights

> Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process. Disabling these predicates should always be a last resort, and is extremely dangerous. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.

Reviewer 2: ```suggestion
Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process. Selectively disabling the predicates is extremely risky, and carries a high chance of either data loss, or causing a cluster to become completely unavailable. This feature is therefore intended exclusively as a troubleshooting mechanism of last resort. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.
```

> Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process. Disabling these predicates should always be a last resort, and is extremely dangerous. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.

Reviewer 2: ```suggestion
Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process. Selectively disabling the predicates is extremely risky, and carries a high chance of either data loss, or causing a cluster to become completely unavailable. This feature is therefore intended exclusively as a troubleshooting mechanism of last resort. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.
```

or something along these lines, I wanted to make the notion stronger that this feature is not intended for every day use.

> Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearcch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process.  Note that disabling of these predicates should be always be a last resort, and are extremely dangerous.  Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.

Reviewer: ```suggestion
Starting with ECK 2.1, the Elasticsearch clusters can have certain upgrade 'predicates' (rules) disabled on a case-by-case basis using annotations on the Elasticsearch custom resource, which allow full control over what rules are considered during the Elasticsearch upgrade process. Disabling these predicates should always be a last resort, and is extremely dangerous. Check the link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-orchestration.html#k8s-advanced-upgrade-control[documentation] for more details.
```
